test/jdk/sun/security/ssl/SSLLogger/LoggingFormatConsistency.java

```diff
@@ -0,0 +1,300 @@
+    private static final String password = "passphrase";
+
+    public static void main(String[] args) throws Exception {
+        if (args.length != 0) {
+            // A non-empty set of arguments occurs when the "runTest" argument
+            // is passed to the test via ProcessTools::executeTestJvm.
+            //
+            // This is done because an OutputAnalyzer is unable to read
+            // the output of the current running JVM, and must therefore create
+            // a test JVM. When this case occurs, it will inherit all specified
+            // JVM properties (keyStore, trustStore, tls protocols, etc.)
+            new LoggingFormatConsistency();
+        } else {
+            // We are in the main JVM that the test is being ran in.
+            var keyStoreFileName = System.getProperty("test.src", "./") + "/" + pathToStores + "/" + keyStoreFile;
+            var trustStoreFileName = System.getProperty("test.src", "./") + "/" + pathToStores + "/" + trustStoreFile;
+
+            // Setting up JVM system properties
+            var keyStoreArg = "-Djavax.net.ssl.keyStore=" + keyStoreFileName;
+            var keyStorePassword = "-Djavax.net.ssl.keyStorePassword=" + password;
+            var trustStoreArg = "-Djavax.net.ssl.trustStore=" + trustStoreFileName;
+            var trustStorePassword = "-Djavax.net.ssl.trustStorePassword=" + password;
+            var testSrc = "-Dtest.src=" + System.getProperty("test.src");
+            var javaxNetDebug = "-Djavax.net.debug=all";
+
+            var correctTlsVersionsFormat = new String[]{"TLSv1", "TLSv1.1", "TLSv1.2", "TLSv1.3"};
+            var incorrectTLSVersionsFormat = new String[]{"TLS10", "TLS11", "TLS12", "TLS13"};
+
+            for (int i = 0; i < correctTlsVersionsFormat.length; i++) {
+                String expectedTLSVersion = correctTlsVersionsFormat[i];
+                String incorrectTLSVersion = incorrectTLSVersionsFormat[i];
+
+                System.out.println("TESTING " + expectedTLSVersion);
+                String activeTLSProtocol = "-Djdk.tls.client.protocols=" + expectedTLSVersion;
+                var output = ProcessTools.executeTestJvm(
+                        testSrc,
+                        keyStoreArg,
+                        keyStorePassword,
+                        trustStoreArg,
+                        trustStorePassword,
+                        activeTLSProtocol,
+                        javaxNetDebug,
+                        "LoggingFormatConsistency",
+                        "runTest"); // Ensuring args.length is greater than 0
+
+                output.shouldContain(expectedTLSVersion);
+                output.shouldNotContain(incorrectTLSVersion);
+            }
+        }
+    }
+
+    /*
+     * Primary constructor, used to drive remainder of the test.
+     *
+     * Fork off the other side, then do your work.
+     */
+    LoggingFormatConsistency() throws Exception {
```

Comment:
Have you looked at the templates available at test/jdk/javax/net/ssl/templates? Check SSLSocketTemplate.java which is using CountDownLatch.

Reply:
I have not! I'll take a look, thanks